alpine-infra/src/main/java/alpine/persistence/AlpineQueryManager.java

```diff
@@ -104,14 +122,17 @@ public ApiKey getApiKey(final String key) {
         if (key.length() != ApiKey.FULL_KEY_LENGTH && key.length() != ApiKey.LEGACY_FULL_KEY_LENGTH) {
             return null;
         }
-        return callInTransaction(() -> {
-            final Query<ApiKey> query = pm.newQuery(ApiKey.class, "publicId == :publicId");
-            query.setParameters(ApiKey.getPublicId(key));
-            ApiKey apiKey = executeAndCloseUnique(query);
-            MessageDigest digest = MessageDigest.getInstance(HASH_METHOD);
-            String hashedKey = HexFormat.of().formatHex(digest.digest(ApiKey.getOnlyKeyAsBytes(key)));
-            return apiKey != null && MessageDigest.isEqual(hashedKey.getBytes(), apiKey.getKey().getBytes()) ? apiKey : null;
-        });
+        ApiKey apiKey = getApiKeyBypublicId(ApiKey.getPublicId(key));
+        MessageDigest digest;
+        try {
+            digest = MessageDigest.getInstance(HASH_METHOD);
+        } catch (NoSuchAlgorithmException e) {
+            LOGGER.warn("This hashing Algorithm is unknow: " + HASH_METHOD);
+            return null;
```

Comment:
Best to throw an `IllegalStateException` here, to make it obvious that something is wrong.